### DISCUSSION TITLE
Purchasing domain

### DISCUSSION THREAD

**Discussion by Danish2023github** [community/community#109895] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hello, everyone.
I am looking forward to buying a domain for my website. Kindly suggest a domain selling website that can be trusted and has some best and affordable domain availability. Also, I would appreciate information about what to keep in mind before purchasing a domain. 

Thank you.

**Comment by DanielWren48** [comment]
Hi @Danish2023github,

A popular and cheap service you can look into is [NameCheap](https://www.namecheap.com/).

Something to keep in mind is that will all those services you are buying a domain for a certain period of time. You'd need to choose for how long you want to own the specific domain for. Keep that in mind when calculating the costs.

**Reply by Danish2023github** [reply]
Thank you for your reply. 
Sometimes, it says that I can purchase a domain for 1 year and it is cheap. But, they say that the same domain will become expensive after one year. Why is that? 

Secondly, I have heard about the security risks on websites. Is there any special security that I am required to purchase?

**Reply by DanielWren48** [reply]
Well, the first year might have some discount or initial lower price to persuade you to buy it.

About the security. Domain registers do not really provide you will any securities by default, and if you want you can buy SSL certificate. (SSL are cryptographic protocols designed to secure communication over a computer network)
But you dont really need it for some basic level applications, and will not be one-click set-up so I'd not bother will that, but its up to you ofc.

**Comment by Danish2023github** [comment]
Also, I have a plan to use a website for business purpose. I want to sell some products on it, but I also want to put some games and news page on it. Would it be suitable to make a website and use it for multiple purposes? Also, the domain name will be the same.

**Comment by Danish2023github** [comment]
And do I need SSL certificate for such website?

2. There is also another thing that I want to know. Sometimes websites get hacked. How do I secure my own website?

**Reply by DanielWren48** [reply]
No one will demand you to have SSL on your website. The purpose for having SSL is to make data exchanged between the user and the server encrypted and secure (important when sensitive information, such as login credentials, personal details, or financial data, is being transmitted).

About hacks. There are multiple ways a website can be attacked, I recommend you [read more about those](https://www.rapid7.com/fundamentals/types-of-attacks) to get a better understanding. The way to secure your own website depends on what you build it with really, if you aim to make a full-stack web app then you most likely will have a [Node.js](https://nodejs.org/en/about) or similar back-end and you will be able to add [middlewares](https://azure.microsoft.com/en-gb/resources/cloud-computing-dictionary/what-is-middleware) to your application to secure it from some of those attacks.

**Comment by Danish2023github** [comment]
There is also one more thing that I would like to add on to my previous question. What is the role of github in website development?

**Reply by DanielWren48** [reply]
Well again, your don't exactly need GitHub to create a web application, GitHub is used for:

- Version control
- Collaboration
- Issue Tracking
- Continuous Integration/Continuous Deployment (CI/CD)

but again depending your your application scale, you are not required  to use GitHub. It is however, beneficial to do so for above mentioned purposes. Can read more about it [here](https://blog.hubspot.com/website/what-is-github-used-for).

**Comment by Danish2023github** [comment]
Now, there is price plan. Can you explain this to me? 

Domain name (Does it mean a free domain name?)
3 Websites
20 GB SSD
30 Mailboxes
Website Builder

<img width="754" alt="hosting price" src="https://github.com/community/community/assets/133471108/3c2560cb-522b-44a8-bda4-5460a7967cda">

**Reply by DanielWren48** [reply]
A domain name is an address that serves as a reference point to access resources on the Internet. github.com, for example, is a domain name for GitHub.

It does not mean its a free domain, the plan includes a domain name. But also keep in mind that you will only be able to choose a domain that is not taken by any one on the internet, so like you will not be able to purchase a github.com for example, because its already owned by GitHub.

- 3/Unlimited Websites - it typically means that you can host and manage multiple separate websites under that single plan.
- GB SSD is how much storage space you will get, kinda like how much stuff you can store on your own SSD/HDD.
- Mailboxes. It means that you have the ability to create and manage a given number of email accounts associated with your domain. 

_A mailbox is essentially an email account associated with your domain. For example, if your domain is "example.com," you can create email accounts like "info@example.com" or "support@example.com."_

**Reply by Danish2023github** [reply]
Wow. So, it would work just like a Gmail account.

**Comment by Danish2023github** [comment]
Now, I also have another question. I also read on the internet about a dynamic and static website. I could not understand it and cannot decide which hosting should I buy.

**Reply by DanielWren48** [reply, marked as answer]
**Static Website:**
A static website is like a fixed, unchanging brochure. Each page is a separate file, and the content doesn't change unless someone manually edits the HTML or other code.
_Example: Imagine a simple website with a few pages that display information about a business, like its services, contact details, and a brief introduction. If you want to update the content, you have to edit the HTML directly._

**Dynamic Website:**
A dynamic website is more like a smart, interactive application. The content can change in real-time based on user input, data from a database, or other factors. The server generates pages on-the-fly.
_Example: Think of a social media platform where your news feed constantly updates with new posts, likes, and comments. The content is not pre-made; it's generated dynamically based on what's happening and who you follow._

Hosting and domains are not the same thing, just to clarify. Hosting is a platform that serves your website to the user, and the domain is just an address. Some popular hosting platform are: 

1. [Hostinger](https://www.hostinger.co.uk/) which also sells domains, so might be worth checking it out.
2. [Netfily](https://www.netlify.com/)
3. [Heroku](https://www.heroku.com/home)
4. [Vercel](https://vercel.com/home)
5. [Render](https://render.com/)

As far as I'm aware, all those platforms allow you to deploy dynamic and static website.

**Reply by Danish2023github** [reply]
Thank you so much once again. You are very helpful. 
So, do I have to specify the host providing websites of my need of either a dynamic or static website? Or I simply purchase the plan and start uploading my website on it. No matter static or dynamic.

**Reply by DanielWren48** [reply]
Well, what kind of application do you mean to deploy, is it any of the projects you have at your github profile?

**Reply by Danish2023github** [reply]
Okay. Let me explain everything to you in detail. So, I need to start a website mainly for business purpose. I am about to start a company. For that purpose, I need a company name and a website as well.

When I decided to choose a company name, it was difficult to choose a domain name because the name I would choose would give me a very expensive domain. I found a solution to that problem. I decided to look for a domain first and then name my company by that domain name. 

My website is mostly going to be related to tech. Basically, computer hardware components, such as SSDs, RAM, and so on. I also want to associate some other functions in that company website. For example, a review page and an email box where I can get emails from the potential buyers. In addition to that product, I also want to dedicate a part of that website to something social, some online games and story pages, some news pages.

Do you think my plan would work well?

**Reply by DanielWren48** [reply]
I mean I'm in no position to say how your business will turn out, but your web application requirements seem reasonable and achievable.
The first part about hardware kinda resembles an e-commerce type of application, the second part about games & social, I don't really know what exactly it is you mean to make but again, no one is restricting you on what you put on your app, so yeah.

Do you know what tech you want to use to create all this with? JavaScript with something like React, or any other languages?

**Reply by Danish2023github** [reply]
Currently, I am looking into Wordpress. I have started learning it. I am learning it online by using xamp app to build the website offline. Also, adding some plugins stuff.

**Reply by DanielWren48** [reply]
I have never done any Wordpress so can't help you there, but as far as I'm aware Wordpress is good for some simple/basic web apps and the description of your project seems a little advance so idk, but again, your decision there.

**Comment by Danish2023github** [comment]
By games I mean webpage related games and story based games. I think many people are not aware of that sort of thing. These used to exist long time ago. Cellufun is an example of websites that provide these online games.

**Reply by Danish2023github** [reply]
I think it would be more static. But, when it comes to the addition of emailbox and review button part, I guess that stuff is called dynamic. Am I right?

**Reply by DanielWren48** [reply]
If you mean the feature where users can leave reviews of your product, then yes, thats dynamic. that also means you need to be saving their reviews in a database, and retrieve them. I have no idea how you achieve this with WordPress tho.

**Comment by Danish2023github** [comment]
Is there anyone live available on these domain and hosting websites for assistance of their buyers?

**Reply by DanielWren48** [reply]
You mean customer support?

**Reply by Danish2023github** [reply]
Yes!

**Reply by DanielWren48** [reply]
All of those hosting services I have mentioned will sure have some sort of customer support.

**Reply by Danish2023github** [reply]
Thank you so much for all this information. I will get back to you for further help.

**Reply by DanielWren48** [reply]
You're welcome.

**Comment by Danish2023github** [comment]
Hi

I recently came across a new problem. This time the problem is about wordpres. I designed a webpage there but the moment i copied the html code, it contained css too. I need separate html and css code files. How do I do that?

Thank you.

**Reply by DanielWren48** [reply]
Hi,
You have your HTML file something like:
```
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Your Page Title</title>
    <link rel="stylesheet" href="styles.css">
</head>
<body>
    <h1>Hello, World!</h1>
    <p>This is a basic HTML boilerplate with linked CSS file.</p>
</body>
</html>
```

and you have to add a link to your css file inside your `<head>` tag in your HTML
```
<link rel="stylesheet" href="styles.css">
```

and have your css file that looks something like:
```
/* styles.css */

/* Resetting default margin and padding */
html, body {
    margin: 0;
    padding: 0;
}

/* Applying some basic styles to headings and paragraphs */
h1 {
    font-size: 24px;
    color: #333;
}

p {
    font-size: 16px;
    color: #666;
    line-height: 1.5;
}
```

In this example the css file should be named `styles.css` as that's how its linked inside the `<head>`, the name is **important**. If you name your css file differently then that the name you should put inside `<head>` when you link it in your HTML.
Also keep your HTML and CSS files in the same directory.

Hope that makes sense.

**Reply by Danish2023github** [reply]
well, the problem is a bit different. 

I have a code that has css inline with html. I want to separate the two codes.

**Reply by Danish2023github** [reply]
I want to separate the two to put it exactly the way you have described above.

**Reply by DanielWren48** [reply]
Well, as far as Im aware there isnt a tool that would separate inline css from html file, so you might need to do that manually.
So if i understand correctly you have your HTML body something like:

```
<body>
    <!-- Content with inline CSS -->
    <div style="background-color: lightblue; padding: 20px;">
        <h1 style="color: darkblue;">Hello, world!</h1>
        <p style="font-family: Arial, sans-serif;">This is an example HTML file with inline CSS.</p>
    </div>
    
    <div style="background-color: lightgreen; padding: 20px;">
        <h2 style="color: darkgreen;">Another section</h2>
        <p style="font-family: Helvetica, sans-serif;">More content here.</p>
    </div>
</body>
```

with inline css to each html element?

If so you can use html classes [read more here](https://www.w3schools.com/html/html_classes.asp) to separate your css into its own file, which will looks something like:
```
<body>
    <!-- Content with CSS classes -->
    <div class="blue-background">
        <h1 class="blue-text">Hello, world!</h1>
        <p class="arial-font">This is an example HTML file with inline CSS.</p>
    </div>
    
    <div class="green-background">
        <h2 class="green-text">Another section</h2>
        <p class="helvetica-font">More content here.</p>
    </div>
</body>
```

and your style.css would use those style classes to apply styles to your html elements and it would look something like:
```
/* CSS classes */
.blue-background {
    background-color: lightblue;
    padding: 20px;
}

.green-background {
    background-color: lightgreen;
    padding: 20px;
}

.blue-text {
    color: darkblue;
}

.green-text {
    color: darkgreen;
}

.arial-font {
    font-family: Arial, sans-serif;
}

.helvetica-font {
    font-family: Helvetica, sans-serif;
}
```

**Reply by Danish2023github** [reply]
Awesome.

**Reply by Danish2023github** [reply]
So, this means I will have to separate these codes for each page.

**Reply by DanielWren48** [reply]
well yeah, it seems like you would, you could of course just keep it the way it is right now but its up to you.

**Reply by Danish2023github** [reply]
It does not really matter even if I keep it this way, right?

**Reply by DanielWren48** [reply]
quick googling, check out this [online tool](http://extractcss.com/) idk if it will do that you need but maybe worth the try

**Reply by DanielWren48** [reply]
no it doesnt matter, if it works the way you need then sure you can just keep it that way

**Reply by Danish2023github** [reply]
i am gonna try the tool you shared. And, if it is fine the way it is then it is good. I really dont like making many files so its better to have a single file for a single webpage.

**Reply by Danish2023github** [reply]
the site to the tool does not work.

**Reply by Danish2023github** [reply]
one more thing. if that webpage contains html and css togther, then this means it also contains javescript. wordpress really should have made the separate files

**Reply by DanielWren48** [reply]
yeah, you can actually have all 3 HTML, CSS, and JavaScript on the same file and it will work fine, I personally do not like keeping them together and would prefer to have each thing separated into its own file, but again whatever works for you...
You can separate JavaScript the same way as css, create a `.js` file and link it in your HTML.

**Reply by Danish2023github** [reply]
I get it. I will give the task of separation of files to my sibling. If that lazy bone sibby wants to do it, this task will definitely be done :) and thanks a lot. You are the best.

**Reply by DanielWren48** [reply]
You're welcome.

**Comment by Danish2023github** [comment]
Hi, Dan.

I have another problem today regarding web development. The problem is related to different screen resolutions. How do I set up my website for a variety of screens where it will be displayed?

For example, if I view it on pc it has a specific resolution, and if I view it on mobile or tablet device, it will have its own resolution. How to set elements' order of display for each screen resolution? 

I am using Wordpress but I would like to know the trick even if it can be achieved through code editing. 

Thank you!

**Reply by Danish2023github** [reply]
And I want to do this test offline from local server.

**Reply by DanielWren48** [reply]
Hey,
Different screens is honestly a pain to deal, with but anyway, with just css you will have to use something called `@media` you can [read more here](https://www.w3schools.com/cssref/css3_pr_mediaquery.php).

So the way this works is you add this `@media` rule to apply different styles for different media types or conditions, such as screen sizes, device orientations, or print vs. screen display. It allows you to specify a block of CSS rules that will only apply under certain conditions.

For example, let's say you want to apply different styles for screens smaller than 600 pixels wide:
```
@media screen and (max-width: 600px) {
  /* Styles for screens smaller than 600px wide */
}
```

You can also combine multiple conditions using logical operators like and and not. For instance:
```
@media screen and (max-width: 600px) and (orientation: portrait) {
  /* Styles for screens smaller than 600px wide in portrait orientation */
}
```
This would apply styles only to screens that are both smaller than 600 pixels wide and in portrait orientation.

And yeah, you will have to do this for all elements you want to scale to some resolution and for each device type you want to cover, desktop, tablet, phone... etc.

You can use `@media `queries along with Flexbox or other layout techniques to change the display order of elements based on different screen sizes or other conditions. 
Here is a cool guide if you want to learn more - https://css-tricks.com/snippets/css/a-guide-to-flexbox/

**Reply by Danish2023github** [reply]
Thank you so much my dear bro. After reading it all, I can feel why you used the word pain in the beginning :D But I will do it.

**Comment by Danish2023github** [comment]
Good day to you, bro. 

I have been thinking about investments at a younger age. Bro, I am worried about future. I know people would say do not worry about the future but focus on the present. But, we need to focus on future, too. 

I was talking to a friend of mine who is old now and is unable to get a job. That friend did not invest in any business or something, such as a passive income. I do not want to end up that way. 

Bro, I have computer. What should I do to earn online? Is there a practical way of doing so without getting certifications?

I have heard that people earn from sites like Patreon and so on. How do they earn?

**Comment by Danish2023github** [comment]
I understand. My friend is a computer specialist. She has no job now.

**Reply by Danish2023github** [reply]
because she is old maybe. this is the reason she told me for her bring jobless. family members are supporting and i think that is an investment for her. but, what if we do not have a family. where are you from ? i am from pakistan.

**Comment by mgunawardhana** [comment]
Sure, here's the formatted text:

---

**Hello!**

When it comes to purchasing a domain for your website, there are a few key considerations. Let’s dive into some recommendations and essential tips:

**Choosing a Trusted Domain Registrar:**

[Namecheap](https://www.namecheap.com/) is a reliable domain registrar that offers affordable domain names starting from $0.99. They manage over 16 million domains, so you’re in good hands.

Other reputable registrars include [Porkbun](https://www.usatoday.com/money/blueprint/business/web-hosting/best-domain-registrar/), [Dynadot](https://www.usatoday.com/money/blueprint/business/web-hosting/best-domain-registrar/), and [IONOS](https://www.forbes.com/advisor/business/software/best-domain-registrar/).

**Tips for Choosing the Perfect Domain Name:**

1. **Select the Right TLD (Top-Level Domain):**
   The TLD is the rightmost part of your domain (like .com, .net, .org). Choose one that aligns with your website’s purpose and audience.

2. **Keep It Simple and Memorable:**
   Opt for a domain name that is easy to spell, short, and memorable. Around 8-12 characters is ideal.

3. **Avoid Doubled Letters, Numbers, and Special Characters:**
   Clarity matters. Steer clear of confusing combinations.

4. **Check for Trademarks:**
   Ensure your chosen domain doesn’t infringe on any existing trademarks.

5. **Include Targeted Keywords:**
   If possible, incorporate relevant keywords related to your business or niche.

6. **Balance Creativity and Availability:**
   Be creative, but also make sure the domain is available.

7. **Make It Brandable:**
   Include your business name to build brand recognition.

8. **Research Social Media Availability:**
   Check if the domain name is available on social media platforms.

9. **Check the Domain’s History:**
   Investigate whether the domain has any negative associations.

10. **Use a Domain Name Generator:**
    Tools like this can spark ideas and suggest available domains.

11. **Relevance to Site Content:**
    Ensure the domain reflects your website’s content or purpose.

12. **Purchase in Advance:**
    Don’t wait until the last minute; secure your domain early.

**Why Choosing the Right Domain Matters:**

- **First Impression:**
  Your domain is often the first thing visitors notice. Make it unique and memorable.

- **Search Engine Optimization (SEO):**
  While not a direct factor, a good domain can improve click-through rates and visitor interest.

- **Branding:**
  Including your business name in the domain helps build brand recognition.

Remember, a well-chosen domain can enhance your website’s credibility and attract more visitors. Happy domain hunting! 🌐🔍

---

**Comment by NooruddinLakhani** [comment]
There is plenty of information online about purchasing a domain, but there are few domains available which are affordable and trusted. I purchased the domain from here [Register Domain](https://register.domains/en/dm/search-domains) It was a good option for me. You can also explore It if you want.
